### ISSUE TITLE
Add countdown timer

### ISSUE THREAD

**Issue by ojasva** [HaikuArchives/BeVexed#34]
Challenges will make the game more interesting.

**Comment by humdingerb** [comment]
IMO the countup timer should be enough. In fact, I'd be pissed I struggled on a hard game for 15 minutes and have it yanked from me close to the end...

**Comment by ojasva** [comment]
I agree with you but here is something I read few days back, I quoting someone here:

> Games are designed around a certain principle. It's called `Flow Theory'. 

>The idea is to balance the skills of the player with the challenge presented, and also, to increase the difficulty of the challenges, as time passes (i.e. as the game Level proceeds, and as Missions [sets of levels] proceed). 

>The goal of the game designer(s) is to keep the `average/typical' player in the `Flow' state (and, thus: engaged, compelled, interested, involved, etc!)
No wonder games are `addictive'; they're designed to be: hugely-rewarding, ie: fun.

So basically, if you complete that hard game in 15 minutes. You get that feeling of "Yeah! I did it it." which is highly rewarding psychologically and you will continue to keep playing more and more. This feature can be optional ofc.

**Comment by owenca** [comment]
@ojasva Then a popup dialog that allows the user to choose whether to continue finishing the game when the timer expires would be user friendly.

**Comment by ojasva** [comment]
@owenca Agreed.
